### ISSUE TITLE
[Bug]: assemble_tops.c produces malformed xml

### ISSUE THREAD

**Issue by mobigroup** [gmtsar/gmtsar#354]
See the example command and XML checker tool output below:
```
PATH=$PATH:/usr/local/GMTSAR/bin assemble_tops 1575 5330 s1a-iw1-slc-vv-20210108t144501-20210108t144528-036048-043974-004 new
xmllint --format new.xml >/dev/null

new.xml:5708: parser error : expected '>'
    <coordinateConversionList count="0"/></coordinateConversionList>
                                                               ^
new.xml:5709: parser error : Opening and ending tag mismatch: product line 2 and coordinateConversion
  </coordinateConversion>
                         ^
new.xml:5710: parser error : Extra content at the end of the document
  <swathMerging>
  ^
```

Original xml:
```
...
  <coordinateConversion>
    <coordinateConversionList count="0"/>
  </coordinateConversion>
  <swathMerging>
    <swathMergeList count="0"/>
  </swathMerging>
</product>
```

Output xml:
```
  <coordinateConversion>
    <coordinateConversionList count="0"/></coordinateConversionList>
  </coordinateConversion>
  <swathMerging>
    <swathMergeList count="0"/></swathMergeList>
  </swathMerging>
</product>
```

Here we see tags "coordinateConversionList" and "swathMergeList" are closed two times ("/></").

**Comment by Xiaohua-Eric-Xu** [comment]
Could be because the count being zero, that triggered some bug in xml.c in GMTSAR.  

Funny thing is that it didn't fail the xml reader. Well it should have. There should be a function called show_tree or something to output the xml structure in the code. 

p.s. I probably don't have time to work on this until weeks later.

**Comment by mobigroup** [comment]
Ok, thanks. I have a set of changes to merge first (UNIX API enhancements and some issues fixes like to printf vs fprintf) to allow easy GMTSAR tool reusing from other programs and scripts.

**Comment by Xiaohua-Eric-Xu** [comment]
adding a fix e0bd2fe837be076a5bc1bacc72dc96a9682aaf62
